### DISCUSSION TITLE
Cant use gatsby after forcing cacheable-request to 10.2.7

### DISCUSSION THREAD

**Discussion by seyoon20087** [gatsbyjs/gatsby#37646] [Help]
# NOTE for users who want to try:
Use this link:
https://gitpod.io/#https://github.com/seyoon20087/gatsby-sample-reproduction/tree/cant-run-due-to-cacheable-request

If you can't use Gitpod (or simply want to try it locally or GitHub Codespaces), you may omit `https://gitpod.io/#` at the beginning of the URL. Make sure you have Node.js 18.x if you use it locally

---

Hello folks,

I use Dependabot on my website. After I forced `cacheable-request` to 10.2.7 with the `resolutions` key in package.json (due to a [Dependabot alert](https://github.com/advisories/GHSA-8x6c-cv3v-vp6g) being generated), I always get the following error on `gatsby develop` or `gatsby build`:

```
yarn run v1.22.19
$ gatsby build
error gatsby build

Build a Gatsby project.

Options:
  --verbose                   Turn on verbose output  [boolean] [default: false]
  --no-color, --no-colors     Turn off the color in output  [boolean] [default: false]
  --json                      Turn on the JSON logger  [boolean] [default: false]
  --prefix-paths              Build site with link paths prefixed with the pathPrefix value in gatsby-config.js. Default is env.PREFIX_PATHS or false.  [boolean] [default: false]
  --no-uglify                 Build site without uglifying JS bundles (for debugging).  [boolean] [default: false]
  --profile                   Build site with react profiling (this can add some additional overhead). See https://reactjs.org/docs/profiler  [boolean] [default: false]
  --graphql-tracing           Trace every graphql resolver, may have performance implications  [boolean] [default: false]
  --open-tracing-config-file  Tracer configuration file (OpenTracing compatible). See https://gatsby.dev/tracing  [string]
  -h, --help                  Show help  [boolean]
  -v, --version               Show the version of the Gatsby CLI and the Gatsby package in the current project  [boolean]
error There was a problem loading the local build command. Gatsby may not be installed. Perhaps you need to run "npm install"? require() of ES Module /home/runner/work/seyoon200[8](https://github.com/seyoon20087/seyoon20087.github.io/actions/runs/4158742479/jobs/7194165344#step:6:9)7.github.io/seyoon20087.github.io/node_modules/cacheable-request/dist/index.js from /home/runner/work/seyoon20087.github.io/seyoon20087.github.io/node_modules/got/dist/source/core/index.js not supported.
Instead change the require of /home/runner/work/seyoon20087.github.io/seyoon20087.github.io/node_modules/cacheable-request/dist/index.js in /home/runner/work/seyoon20087.github.io/seyoon20087.github.io/node_modules/got/dist/source/core/index.js to a dynamic import() which is available in all CommonJS modules.


  Error:[ERR_REQUIRE_ESM]: require() of ES Module /home/runner/work/seyoon20087  .github.io/seyoon20087.github.io/node_modules/cacheable-request/dist/index.js   from /home/runner/work/seyoon20087.github.io/seyoon20087.github.io/node_module  s/got/dist/source/core/index.js not supported.
  Instead change the require of /home/runner/work/seyoon20087.github.io/seyoon20  087.github.io/node_modules/cacheable-request/dist/index.js in /home/runner/wor  k/seyoon20087.github.io/seyoon20087.github.io/node_modules/got/dist/source/cor  e/index.js to a dynamic import() which is available in all CommonJS modules.
  
  - index.js:13 Object.<anonymous>
    [seyoon20087.github.io]/[got]/dist/source/core/index.js:13:26
  
  - types.js:14 Object.<anonymous>
    [seyoon20087.github.io]/[got]/dist/source/as-promise/types.js:14:16
  
  - index.js:16 Object.<anonymous>
    [seyoon20087.github.io]/[got]/dist/source/as-promise/index.js:16:17
  
  - create.js:15 Object.<anonymous>
    [seyoon20087.github.io]/[got]/dist/source/create.js:15:22
  
  - index.js:14 Object.<anonymous>
    [seyoon20087.github.io]/[got]/dist/source/index.js:14:18
  
  - start-server.js:8 Object.<anonymous>
    [seyoon20087.github.io]/[gatsby]/dist/utils/start-server.js:8:35
  
  - start-webpack-server.js:14 Object.<anonymous>
    [seyoon20087.github.io]/[gatsby]/dist/services/start-webpack-server.js:14:20  
  - index.js:25 Object.<anonymous>
    [seyoon20087.github.io]/[gatsby]/dist/services/index.js:25:27
  
  - index.js:[9](https://github.com/seyoon20087/seyoon20087.github.io/actions/runs/4158742479/jobs/7194165344#step:6:10) Object.<anonymous>
    [seyoon20087.github.io]/[gatsby]/dist/bootstrap/index.js:9:17
  
  - build.js:12 Object.<anonymous>
    [seyoon20087.github.io]/[gatsby]/dist/commands/build.js:12:18
  
  - create-cli.js:62 resolveLocalCommand
    [seyoon20087.github.io]/[gatsby-cli]/lib/create-cli.js:62:19
  
  - create-cli.js:86 
    [seyoon20087.github.io]/[gatsby-cli]/lib/create-cli.js:86:24
  
  - create-cli.js:23 Object.handler
    [seyoon20087.github.io]/[gatsby-cli]/lib/create-cli.js:23:19
  
  - command.js:196 Object.runCommand
    [seyoon20087.github.io]/[yargs]/build/lib/command.js:196:48
  
  - yargs.js:[10](https://github.com/seyoon20087/seyoon20087.github.io/actions/runs/4158742479/jobs/7194165344#step:6:11)43 Object.parseArgs [as _parseArgs]
    [seyoon20087.github.io]/[yargs]/build/lib/yargs.js:1043:55
  
  - yargs.js:584 Object.parse
    [seyoon20087.github.io]/[yargs]/build/lib/yargs.js:584:29
  
  - create-cli.js:477 createCli
    [seyoon20087.github.io]/[gatsby-cli]/lib/create-cli.js:477:[13](https://github.com/seyoon20087/seyoon20087.github.io/actions/runs/4158742479/jobs/7194165344#step:6:14)6
  
  - index.js:67 Object.<anonymous>
    [seyoon[20](https://github.com/seyoon20087/seyoon20087.github.io/actions/runs/4158742479/jobs/7194165344#step:6:21)087.github.io]/[gatsby-cli]/lib/index.js:67:[26](https://github.com/seyoon20087/seyoon20087.github.io/actions/runs/4158742479/jobs/7194165344#step:6:27)
  
  - gatsby.js:4 Object.<anonymous>
    [seyoon20087.github.io]/[gatsby]/dist/bin/gatsby.js:4:1
  
  - cli.js:3 Object.<anonymous>
    [seyoon20087.github.io]/[gatsby]/cli.js:3:1
  

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I do not have a way of fixing this on my part (and I do not want to ignore any Dependabot warnings, nor downgrade).

Maybe you are not inling dependencies on `gatsby` (and instead rely on the `node_modules` mechanism -- see https://overreacted.io/npm-audit-broken-by-design/#what-next -> check 'inline all dependencies during publish' section)?
Some build tools like [Next.js](https://github.com/vercel/next.js/tree/v13.1.7-canary.10/packages/next/src/compiled) already do that and Gatsby may want to try that instead.

**Comment by LekoArts** [comment]
While those audit messages might seem scary they are almost always irrelevant in the context of Gatsby as it's a build tool. So there is nothing to "fix" as the warning is not relevant.

Gatsby creates static assets and runs everything at build time and not during runtime. `npm audit` is designed for runtime / Node apps so it flags issues that can occur there. This means that almost every "vulnerability" report we receive are false positives. While in principle Gatsby can also have vulnerabilities you need to make sure that it's relevant to Gatsby before reporting it. For example a "Regex DDOS attack" can never be a real vulnerability for a development-time tool. If you want to override/update a transitive dependency you can use [yarn resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/).

You can also read [npm audit: Broken by Design](https://overreacted.io/npm-audit-broken-by-design/) to learn more about this.

If you know that a vulnerability affects Gatsby because you understand what the vulnerability is, please report it [through our security document](https://github.com/gatsbyjs/gatsby/blob/master/SECURITY.md). Thanks!

**Reply by seyoon20087** [reply]
In the link I provided ( https://overreacted.io/npm-audit-broken-by-design/#what-next ) it said:
> [...]
> Inline all dependencies during publish. This is what I’m increasingly seeing packages similar to Create React App do. For example, both [Vite](https://unpkg.com/browse/vite@2.4.1/dist/node/) and [Next.js](https://unpkg.com/browse/next@11.0.1/dist/) simply bundle their dependencies directly in the package instead of relying on the npm node_modules mechanism. From a maintainer’s point of view, [the upsides are clear](https://github.com/vitejs/vite/blob/main/.github/contributing.md#notes-on-dependencies): you get faster boot time, smaller downloads, and — as a nice bonus — **no bogus vulnerability reports from your users**. It’s a neat way to game the system but I’m worried about the incentives npm is creating for the ecosystem. Inlining dependencies kind of goes against the whole point of npm.
> [...]

Just a quick question - you don't inline the dependency like Next.js does ?

Also note that I am using Dependabot and it will always emit warnings in my repo (even I put it in `devDependencies` key in my package.json)
